src/app/child-dev-project/notes/note-details/note-details.component.ts

```diff
@@ -41,84 +45,83 @@ import { MatDialogModule } from "@angular/material/dialog";
-  readonly dateLabel = Note.schema.get("date").label;
-  readonly statusLabel = Note.schema.get("warningLevel").label;
-  readonly categoryLabel = Note.schema.get("category").label;
-  readonly authorsLabel = Note.schema.get("authors").label;
-  readonly authorsPlaceholder = this.getPlaceholder(this.authorsLabel);
-  readonly subjectLabel = Note.schema.get("subject").label;
-  readonly textLabel = Note.schema.get("text").label;
-  readonly childrenLabel = Note.schema.get("children").label;
-  readonly childrenPlaceholder = this.getPlaceholder(this.childrenLabel);
-  readonly schoolsLabel = Note.schema.get("schools").label;
-  readonly schoolsPlaceholder = this.getPlaceholder(this.schoolsLabel);
-  readonly relatedEntitiesSchema: EntitySchemaField =
-    Note.schema.get("relatedEntities");
-
-  readonly INTERACTION_TYPE_CONFIG = INTERACTION_TYPE_CONFIG_ID;
-  readonly compareFn = compareEnums;
-  includeInactiveChildren: boolean = false;
 
   /** export format for notes to be used for downloading the individual details */
   exportConfig: ExportColumnConfig[];
 
-  /** Is it mobile view or not */
-  mobile = false;
+  topForm = ["date", "warningLevel", "category", "authors"].map((field) => [
+    toFormFieldConfig(field),
+  ]);
+  middleForm = ["subject", "text"].map(toFormFieldConfig);
+  bottomForm = ["children", "schools"].map(toFormFieldConfig);
+  form: EntityForm<Note>;
+  tmpEntity: Note;
 
   constructor(
     private configService: ConfigService,
-    private screenWidthObserver: ScreenWidthObserver
+    private entityFormService: EntityFormService,
+    @Inject(MAT_DIALOG_DATA) data: { entity: Note },
+    private dialog: MatDialogRef<any>,
+    private alertService: AlertService,
+    private entityRemoveService: EntityRemoveService
   ) {
+    this.entity = data.entity;
     this.exportConfig = this.configService.getConfig<{
       config: EntityListConfig;
     }>("view:note").config.exportConfig;
-    this.screenWidthObserver
-      .platform()
-      .pipe(untilDestroyed(this))
-      .subscribe((isDesktop) => (this.mobile = !isDesktop));
+    const formConfig = this.configService.getConfig<any>(
+      "app-config:note-details"
+    );
+    ["topForm", "middleForm", "bottomForm"].forEach((form) => {
+      this[form] = formConfig?.[form]?.map(toFormFieldConfig) ?? this[form];
+    });
   }
 
-  toggleIncludeInactiveChildren() {
-    this.includeInactiveChildren = !this.includeInactiveChildren;
-    // This needs to be set so that the filtering will start immediately
-    this.filterInactiveChildren = this.includeInactiveChildren
-      ? (_) => true
-      : (c) => c.isActive;
+  ngOnInit() {
+    this.form = this.entityFormService.createFormGroup(
+      this.middleForm.concat(...this.topForm, this.bottomForm),
+      this.entity
+    );
+    this.tmpEntity = this.entity.copy();
+    this.form.valueChanges.pipe(untilDestroyed(this)).subscribe((value) => {
+      this.tmpEntity = Object.assign(this.tmpEntity, value);
+    });
```

Comment:
why do we create and update this tmpEntity? Only for the title in the template?

Reply:
Yes, don't know how necessary it is

Reply:
```suggestion
    // create an object reflecting unsaved changes to use in template (e.g. for dynamic title)
    this.tmpEntity = this.entity.copy();
    this.form.valueChanges.pipe(untilDestroyed(this)).subscribe((value) => {
      this.tmpEntity = Object.assign(this.tmpEntity, value);
    });
```